test/integ/test_connection.py

```diff
@@ -131,6 +131,63 @@ def test_with_config(db_parameters):
         cnx.close()
 
 
+@pytest.mark.skipolddriver
+def test_with_tokens(db_parameters):
+    """Creates a connection using session and master token."""
+    initial_cnx = snowflake.connector.connect(
+        user=db_parameters["user"],
+        password=db_parameters["password"],
+        host=db_parameters["host"],
+        port=db_parameters["port"],
+        account=db_parameters["account"],
+        protocol=db_parameters["protocol"],
+        timezone="UTC",
+    )
+    assert initial_cnx, "invalid initial cnx"
+    master_token = initial_cnx._rest._master_token
+    session_token = initial_cnx._rest._token
+    token_cnx = snowflake.connector.connect(
+        account=db_parameters["account"],
+        host=db_parameters["host"],
+        port=db_parameters["port"],
+        protocol=db_parameters["protocol"],
+        session_token=session_token,
+        master_token=master_token,
+    )
+    assert token_cnx, "invalid second cnx"
+    initial_cnx.close()
+    token_cnx.close()
+
+
+@pytest.mark.skipolddriver
+def test_with_tokens_expired(db_parameters):
+    """Creates a connection using session and master token."""
+    initial_cnx = snowflake.connector.connect(
+        user=db_parameters["user"],
+        password=db_parameters["password"],
+        host=db_parameters["host"],
+        port=db_parameters["port"],
+        account=db_parameters["account"],
+        protocol=db_parameters["protocol"],
+        timezone="UTC",
+    )
+    assert initial_cnx, "invalid initial cnx"
+    master_token = initial_cnx._rest._master_token
+    session_token = initial_cnx._rest._token
+    initial_cnx.close()
+
+    with pytest.raises(Error):
```

Comment:
is this throwing the `InterfaceError` based upon our previous discussion?

> snowflake.connector.errors.InterfaceError: 250003 (08001): 401 Unauthorized: post https://ig86763.preprod12.us-west-2.aws.snowflakecomputing.com:443/session/heartbeat?requestId=4a63f07e-4928-4f98-b02d-58a4e7151f73&request_guid=148cac21-a4da-4341-bb24-862f8a6211c7

Reply:
It is now a ProgrammingError--I have made the test be more specific and changed it to ProgrammingError.